build.cake

```diff
@@ -0,0 +1,50 @@
+var target = Argument("target", "Default");
+var configuration = Argument("configuration", "Release");
+
+var artifactsDir = "./artifacts/";
+var projectFile = "./src/Grok.Net/Grok.Net.csproj";
+var binDir = "./src/Grok.Net/bin/";
+var objDir = "./src/Grok.Net/obj/";
+
+Task("Clean")
+    .Does(() =>
+{
+    CleanDirectories(new[]{ artifactsDir, binDir, objDir });
+});
+
+Task("Restore-NuGet-Packages")
+    .IsDependentOn("Clean")
+    .Does(() =>
+{
+    DotNetCoreRestore(projectFile);
+});
+
+Task("Build")
+    .IsDependentOn("Restore-NuGet-Packages")
+    .Does(() =>
+{
+    DotNetCoreBuild(projectFile, new DotNetCoreBuildSettings(){ Configuration = configuration });
```

Comment:
By default `DotNetCoreBuild` will run nuget restore command. As it is currently implemented, we run  **"Restore-NuGet-Packages"** task before **"Build"**. So, we don't actually need to run restore command again. You can do it in this way:
```csharp
DotNetCoreBuild(projectFile, new DotNetCoreBuildSettings
{
	Configuration = configuration,
	NoRestore = true
});
```

it will decrease build time. yes it is only a few milliseconds, but sometimes milliseconds is important :)

Reply:
A fair point. Changed.